lib/SPIRV/SPIRVReader.cpp

```diff
@@ -1760,6 +1760,40 @@ Value *SPIRVToLLVM::transValueWithoutDecoration(SPIRVValue *BV, Function *F,
   case OpGetKernelNDrangeSubGroupCount:
     return mapValue(
         BV, transSGSizeQueryBI(static_cast<SPIRVInstruction *>(BV), BB));
+  case OpFPGARegINTEL: {
+    IRBuilder<> Builder(BB);
+
+    SPIRVFPGARegINTELInstBase *BC =
+        static_cast<SPIRVFPGARegINTELInstBase *>(BV);
+
+    Type *Int8PtrTyPrivate = Type::getInt8PtrTy(*Context, SPIRAS_Private);
+    IntegerType *Int32Ty = Type::getInt32Ty(*Context);
+
+    Value *UndefInt8Ptr = UndefValue::get(Int8PtrTyPrivate);
+    Value *UndefInt32 = UndefValue::get(Int32Ty);
+
+    StringRef AnnotStr("__builtin_intel_fpga_reg");
+    Constant *GS = BB->getModule()->getNamedGlobal("__reg");
```

Comment:
Why "__reg" ?

Reply:
As a submitter, I could speculate that this is to avoid possible collisions with the already-initialized named globals. @asavonic, could you please confirm/correct me here?

Reply:
Not sure I understand either. 
What if we already happen to have a global named "__reg" with a different content? This code is going to use this totally unrelated global. If that is an attempt to cache string constants, then IRBuilder.CreateGlobalStringPtr should do this.

Reply:
This was a very dangerous workaround that would produce ill code should the variable already exist